compiler/rustc_resolve/src/def_collector.rs

```diff
@@ -128,6 +128,10 @@ impl<'a, 'b, 'tcx> DefCollector<'a, 'b, 'tcx> {
 }
 
 impl<'a, 'b, 'tcx> mut_visit::MutVisitor for DefCollector<'a, 'b, 'tcx> {
+    fn visit_span(&mut self, span: &mut Span) {
+        *span = span.with_parent(Some(self.parent_def));
```

Comment:
As I understand all the old spans here should have no parent, maybe add an assert?
(Can be turned into a debug assert before merging.)